app/components/Authentication/Github/Github.jsx

```diff
@@ -1,248 +1,513 @@
             })
-            .catch((err) => {
-                console.error(err);
-            });
     }
 
-    submitGithubPolicy() {
-        callVaultApi('post', `auth/github/map/teams/${this.state.teamName}`, null, { value: this.state.checkedPolicies }, null)
-            .then((resp) => {
-
+    createUpdateConfig() {
+        callVaultApi('post', `${this.state.baseVaultPath}/config`, null, this.state.newConfig)
+            .then(() => {
+                snackBarMessage(`Backend ${this.state.baseVaultPath}/config has been updated`);
+                this.setState({ isBackendConfigured: true, config: this.state.newConfig });
             })
-            .catch((err) => {
-                this.setState({ errorMessage: err });
-            });
+            .catch(snackBarMessage);
     }
 
-    getTeamPolicy(teamName) {
-        let uri = `auth/github/map/teams/${this.state.teamName}`;
-        callVaultApi('get', uri, null, null, null)
-            .then((resp) => {
-                let policyNames = _.get(resp, "data.data.value").split(',');
-
-                _.forEach(this.state.policies, (policy) => {
-                    policy.checked = (policyNames.indexOf(policy.name) > -1);
-                });
-                this.setState({
-                    policies: this.state.policies
-                });
+    createUpdateRole(id) {
+        let itemId = this.state.selectedTab;
+        tokenHasCapabilities(['create', 'update'], `${this.state.baseVaultPath}/map/${itemId}s/${id}`)
+            .then(() => {
+                let updateObj = _.clone(this.state.itemConfig);
+                //updateObj.policies = updateObj.policies.join(',');
+                callVaultApi('post', `${this.state.baseVaultPath}/map/${itemId}s/${id}`, null, updateObj)
+                    .then(() => {
+                        snackBarMessage(`GitHub ${this.state.selectedTab} ${id} has been updated`);
+                        this.listGithubTeams();
+                        this.listGithubUsers();
+                        this.setState({ openItemDialog: false, openNewItemDialog: false, itemConfig: _.clone(this.itemConfigSchema), selectedItemId: '' });
+                        browserHistory.push(this.state.baseUrl);
+                    })
+                    .catch(snackBarMessage);
+            })
+            .catch(() => {
+                this.setState({ selectedRoleId: '' })
+                snackBarMessage(new Error(`No permissions to display properties for role ${id}`));
             })
-            .catch((err) => {
-                this.setState({ errorMessage: `${err} - URI: ${decodeURI(uri)}` });
-            });
     }
 
-    teamNameChanged(e, v) {
-        if (v === '') {
-            _.forEach(this.state.policies, (policy) => {
-                policy.checked = false;
-            });
-
-            this.setState({
-                policies: this.state.policies
-            });
+    componentDidMount() {
+        if (this.props.params.splat) {
```

Comment:
Splat, i believe, comes from ruby, which is a type of operator that contains the `*` values after a given endpoint to match. So you could be telling your server to respond to `example.com/hello/world/*` where the * is the splat. So when you get `example.com/hello/world/vault-ui`, the splat in this case would contain `vault-ui`. This is how i understand it, which is probably not 100%, that frontend guy @Lucretius should have more details.